x-pack/plugins/security_solution/public/timelines/components/side_panel/event_details/use_get_alert_details_flyout_link.ts

```diff
@@ -20,13 +21,16 @@ export const useGetAlertDetailsFlyoutLink = ({
   timestamp: string;
 }) => {
   const { getAppUrl } = useAppUrl();
+  const alertDetailPath = buildAlertDetailPath({ alertId: _id, index: _index, timestamp });
+  const isPreviewAlert = _index.includes(DEFAULT_PREVIEW_INDEX);
+
   // getAppUrl accounts for the users selected space
   const alertDetailsLink = useMemo(() => {
-    const url = getAppUrl({
-      path: `${ALERTS_PATH}/${_id}?index=${_index}&timestamp=${timestamp}`,
-    });
+    if (isPreviewAlert) return null;
+    const url = getAppUrl({ path: alertDetailPath });
+    // We use window.location.origin instead of http.basePath as the http.basePath has to be configured in config dev yml
```

Comment:
While we expect `publicBaseUrl` to be configured by most of our users, it's not guaranteed to always be configured. Since we have guaranteed access to the location.origin on the UI, we use that in place of `core.http.publicBaseUrl`,

Reply:
I desk tested this with a specific alert, in a non-default space, and verified that:

✅ when Kibana is run without a base path, a valid link (with no base path) is copied:

```
http://localhost:5601/s/space-madness/app/security/alerts/redirect/8c6f83aeefa05a130769c3bf2419d6923714554916ce4b990966c7b9d1459261?index=.alerts-security.alerts-space-madness&timestamp=2023-04-24T20%3A54%3A25.941Z
```

✅ when Kibana is run with a base path, a valid link for the same alert (that includes the base path) is copied:

```
http://localhost:5601/jgy/s/space-madness/app/security/alerts/redirect/8c6f83aeefa05a130769c3bf2419d6923714554916ce4b990966c7b9d1459261?index=.alerts-security.alerts-space-madness&timestamp=2023-04-24T20%3A54%3A25.941
```